tests/MongoDB.Driver.Tests/Specifications/transactions/TransactionTestRunner.cs

```diff
@@ -54,21 +57,34 @@ public sealed class TransactionTestRunner
 
         // private fields
         private string _databaseName = "transaction-tests";
+        private readonly List<IDisposable> _disposables = new List<IDisposable>();
         private string _collectionName = "test";
 
         // public methods
+        public void ConfigureFailPoint(IServer server, ICoreSessionHandle session, BsonDocument failCommand)
+        {
+            var failPoint = FailPoint.Configure(server, session, failCommand);
+            _disposables.Add(failPoint);
+        }
+
+        public async Task ConfigureFailpointAsync(IServer server, ICoreSessionHandle session, BsonDocument failCommand)
+        {
+            var failPoint = await Task.Run(() => FailPoint.Configure(server, session, failCommand)).ConfigureAwait(false);
+            _disposables.Add(failPoint);
+        }
+
+        public void Dispose()
+        {
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
         [SkippableTheory]
         [ClassData(typeof(TestCaseFactory))]
         public void Run(JsonDrivenTestCase testCase)
         {
-            if (CoreTestConfiguration.Cluster.Description.Type == ClusterType.Sharded)
-            {
-                RequireServer.Check().Supports(Feature.ShardedTransactions);
-            } 
-            else
-            {
-                RequireServer.Check().Supports(Feature.Transactions).ClusterType(ClusterType.ReplicaSet);
-            }
+            var allowedTopologies = testCase.Shared.GetValue("topology", defaultValue: new BsonArray()).AsBsonArray;
+            CheckClusterTopology(allowedTopologies);
```

Comment:
Let's move this Check to the other Run method where we already have shared and test split out.